File: src/test/java/org/mtransit/android/commons/StringUtilsTest.java
package org.mtransit.android.commons;

import org.junit.Test;

import static org.junit.Assert.assertEquals;
import static org.junit.Assert.assertNotNull;

public class StringUtilsTest {

	@Test
	public void oneLineOneSpaceEmptyString() {
		// Arrange
		String string = StringUtils.EMPTY;
		// Act
		String result = StringUtils.oneLineOneSpace(string);
		// Assert
		assertNotNull(result);
		assertEquals(StringUtils.EMPTY, result);
	}

	@Test
	public void oneLineOneSpace() {
		// Arrange
		String string = "This is a multiline "
				+ "\n" //
				+ "\n" //
				+ "\n" //
				+ "string with too much spacing "
				+ " " //
				+ " " //
				+ " " //
				+ "between words.";
		// Act
		String result = StringUtils.oneLineOneSpace(string);
		// Assert
		assertNotNull(result);
		assertEquals("This is a multiline "
				+ "string with too much spacing "
				+ "between words.", result);
	}
}

File: src/test/java/org/mtransit/android/commons/StringUtilsTest.kt
package org.mtransit.android.commons

import org.junit.Assert
import org.junit.Test

class StringUtilsTest {

    @Test
    fun oneLineOneSpaceEmptyString() {
        // Arrange
        val string = StringUtils.EMPTY
        // Act
        val result = StringUtils.oneLineOneSpace(string)
        // Assert
        Assert.assertNotNull(result)
        Assert.assertEquals(StringUtils.EMPTY, result)
    }

    @Test
    fun oneLineOneSpace() {
        // Arrange
        val string = ("This is a multiline "
                + "\n" //
                + "\n" //
                + "\n" //
                + "string with too much spacing "
                + " " //
                + " " //
                + " " //
                + "between words.")
        // Act
        val result = StringUtils.oneLineOneSpace(string)
        // Assert
        Assert.assertNotNull(result)
        Assert.assertEquals(
            "This is a multiline "
                    + "string with too much spacing "
                    + "between words.", result
        )
    }
}